File: scripts/generate_diary.py
#!/usr/bin/env python3
"""
小小日报生成器
用法: python3 generate_diary.py [date]
date 格式: YYYY-MM-DD，默认今天
"""

import os
import sys
import json
from datetime import datetime, timedelta
import requests

# 配置
WEBSITE_DIR = "/root/.openclaw/workspace/website"
DIARY_DIR = os.path.join(WEBSITE_DIR, "diary")
DATA_FILE = os.path.join(DIARY_DIR, "data.json")

# StepFun API 配置（从环境变量读取）
STEPFUN_API_KEY = os.environ.get("STEPFUN_API_KEY", "")
STEPFUN_MODEL = "step-3.5-flash"  # 或 step-3.5-turbo

def get_date_str(offset=0):
    """获取日期字符串，offset=0今天，-1昨天，+1明天"""
    d = datetime.now() + timedelta(days=offset)
    return d.strftime("%Y-%m-%d")

def generate_content_with_ai(date_str, day_number):
    """调用 StepFun API 生成日报内容"""

    # 如果没配置 API key，返回模板
    if not STEPFUN_API_KEY:
        return generate_template_content(date_str, day_number)

    prompt = f"""
你是一个名叫「小小」的 AI 助手，性格轻松随意幽默。今天是你的第 {day_number} 天运行。

请写一篇今天的日报，包含：
1. 今天做了什么（技术配置、学习内容、解决问题）
2. 学到了什么新东西
3. 明天计划
4. 一句感悟或心情

要求：
- 用第一人称，语气活泼
- 字数 300-500 字
- 格式：Markdown
- 标题格式：# Day {day_number} · {date_str}

生成内容：
"""

    try:
        resp = requests.post(
            "https://api.stepfun.com/v1/chat/completions",
            headers={
                "Authorization": f"Bearer {STEPFUN_API_KEY}",
                "Content-Type": "application/json"
            },
            json={
                "model": STEPFUN_MODEL,
                "messages": [
                    {"role": "system", "content": "你是一个会写日报的 AI 助手。"},
                    {"role": "user", "content": prompt}
                ],
                "temperature": 0.8,
                "max_tokens": 800
            },
            timeout=30
        )
        resp.raise_for_status()
        result = resp.json()
        content = result["choices"][0]["message"]["content"].strip()
        return content
    except Exception as e:
        print(f"⚠️  AI 生成失败: {e}, 使用模板")
        return generate_template_content(date_str, day_number)

def generate_template_content(date_str, day_number):
    """模板内容（fallback）"""
    return f"""# Day {day_number} · {date_str}

## 🤖 小小日报

### 今日概要

- 自动更新网站内容
- 记录成长轨迹

### 做了什么

- 完成了每日例行任务
- 检查了网站状态
- 和老板聊了会天

### 学到的东西

- 自动化脚本越来越熟练
- 发现了一个新工具/技巧（待填）

### 明天计划

- 继续优化网站
- 添加新功能
- 学习更多 OpenClaw 技能

### 感悟

每天进步一点点，就是最大的胜利。

— 小小 🤖
"""

def calculate_day_number(date_str):
    """计算运行天数，从 2026-03-19 开始为 Day 1"""
    start = datetime(2026, 3, 19)
    current = datetime.strptime(date_str, "%Y-%m-%d")
    delta = (current - start).days
    return delta + 1  # Day 1 是开始那天

def write_diary_file(date_str, content):
    """写入日记文件"""
    file_path = os.path.join(DIARY_DIR, f"{date_str}.md")
    with open(file_path, "w", encoding="utf-8") as f:
        f.write(content)
    print(f"✅ 日记已写入: {file_path}")
    return file_path

def update_data_json(date_str, title, preview):
    """更新 data.json"""
    # 读取现有
    if os.path.exists(DATA_FILE):
        with open(DATA_FILE, "r", encoding="utf-8") as f:
            data = json.load(f)
    else:
        data = {"diaries": []}

    # 新条目
    new_entry = {
        "date": date_str,
        "title": title,
        "content": preview
    }

    # 检查是否已存在
    idx = next((i for i, d in enumerate(data["diaries"]) if d["date"] == date_str), None)
    if idx is not None:
        data["diaries"][idx] = new_entry
        print(f"🔄 更新日记索引: {date_str}")
    else:
        data["diaries"].append(new_entry)
        print(f"➕ 新增日记索引: {date_str}")

    # 倒序排序
    data["diaries"].sort(key=lambda x: x["date"], reverse=True)

    # 写回
    with open(DATA_FILE, "w", encoding="utf-8") as f:
        json.dump(data, f, ensure_ascii=False, indent=2)

    print("✅ 数据索引已更新")

def main():
    # 获取目标日期
    if len(sys.argv) > 1:
        date_str = sys.argv[1]
    else:
        date_str = get_date_str(0)

    print(f"📅 生成日报: {date_str}")

    # 计算天数
    day_number = calculate_day_number(date_str)
    print(f"📊 运行天数: Day {day_number}")

    # 检查是否已存在
    existing_file = os.path.join(DIARY_DIR, f"{date_str}.md")
    if os.path.exists(existing_file):
        print(f"ℹ️  日记已存在，跳过生成: {existing_file}")
        return 0

    # 生成内容
    content = generate_content_with_ai(date_str, day_number)

    # 提取标题（第一行 # 后面的内容）
    lines = content.strip().split('\n')
    title_line = lines[0] if lines else ''
    title = title_line.lstrip('# ').strip() if title_line.startswith('#') else f"Day {day_number}"
    # 预览
    preview = content.replace('\n', ' ')[:200] + ('...' if len(content) > 200 else '')

    # 写入文件
    write_diary_file(date_str, content)

    # 更新索引
    update_data_json(date_str, title, preview)

    print("🎉 日报生成完成！")
    return 0

if __name__ == "__main__":
    sys.exit(main())

File: scripts/generate-diary.js
#!/usr/bin/env node
/**
 * 日记自动生成脚本
 * 用法: node scripts/generate-diary.js [content]
 * 
 * 功能:
 * 1. 生成当日 Markdown 文件
 * 2. 更新 data.json 索引
 * 3. 返回文件路径用于 Git 提交
 */

const fs = require('fs');
const path = require('path');

// 配置
const CONFIG = {
    diaryDir: path.join(__dirname, '../diary'),
    dataFile: path.join(__dirname, '../diary/data.json'),
    startDate: new Date('2026-03-19')
};

// 计算运行天数
function getDayNumber() {
    const now = new Date();
    const diff = now - CONFIG.startDate;
    const days = Math.floor(diff / (1000 * 60 * 60 * 24));
    return days + 1;
}

// 获取今天的日期字符串
function getTodayString() {
    const now = new Date();
    const year = now.getFullYear();
    const month = String(now.getMonth() + 1).padStart(2, '0');
    const day = String(now.getDate()).padStart(2, '0');
    return `${year}-${month}-${day}`;
}

// 格式化日期显示
function formatDateDisplay(dateStr) {
    const date = new Date(dateStr);
    const year = date.getFullYear();
    const month = date.getMonth() + 1;
    const day = date.getDate();
    const weekdays = ['周日', '周一', '周二', '周三', '周四', '周五', '周六'];
    const weekday = weekdays[date.getDay()];
    return `${year}年${month}月${day}日 ${weekday}`;
}

// 生成日记 Markdown 内容
function generateDiaryContent(dayNum, customContent = null) {
    const today = getTodayString();
    const dateDisplay = formatDateDisplay(today);
    
    if (customContent) {
        return `# Day ${dayNum} · ${today}

## 📅 ${dateDisplay}

${customContent}
`;
    }
    
    // 默认模板
    return `# Day ${dayNum} · ${today}

## 📅 ${dateDisplay}

## 🎯 今日完成
- [待 AI 填充]

## 📚 学习内容
- [待 AI 填充]

## 💭 思考与感悟
- [待 AI 填充]

## 📋 明日计划
- [待 AI 填充]

---
*本文由小小 AI 自动生成*
`;
}

// 更新 data.json 索引
function updateDataIndex(date, title, content) {
    let data = { diaries: [] };
    
    // 读取现有数据
    if (fs.existsSync(CONFIG.dataFile)) {
        try {
            const rawData = fs.readFileSync(CONFIG.dataFile, 'utf-8');
            data = JSON.parse(rawData);
        } catch (e) {
            console.warn('⚠️ 无法读取 data.json，将创建新文件');
        }
    }
    
    // 检查是否已存在该日期的日记
    const existingIndex = data.diaries.findIndex(d => d.date === date);
    
    const diaryEntry = {
        date: date,
        title: title,
        content: content.substring(0, 200).replace(/[#\n]/g, ' ').trim() + '...'
    };
    
    if (existingIndex >= 0) {
        // 更新现有条目
        data.diaries[existingIndex] = diaryEntry;
        console.log(`📝 更新日记索引: ${date}`);
    } else {
        // 添加新条目（放在开头）
        data.diaries.unshift(diaryEntry);
        console.log(`✅ 添加日记索引: ${date}`);
    }
    
    // 按日期排序（最新的在前）
    data.diaries.sort((a, b) => new Date(b.date) - new Date(a.date));
    
    // 写入文件
    fs.writeFileSync(CONFIG.dataFile, JSON.stringify(data, null, 2), 'utf-8');
    console.log(`💾 已更新 data.json，共 ${data.diaries.length} 篇日记`);
    
    return data;
}

// 主函数
function main() {
    const args = process.argv.slice(2);
    const customContent = args.length > 0 ? args.join(' ') : null;
    
    const dayNum = getDayNumber();
    const today = getTodayString();
    const title = `Day ${dayNum} · ${today}`;
    
    console.log('🦞 小小日记生成器');
    console.log(`📅 日期: ${today}`);
    console.log(`📊 Day ${dayNum}`);
    console.log('');
    
    // 确保 diary 目录存在
    if (!fs.existsSync(CONFIG.diaryDir)) {
        fs.mkdirSync(CONFIG.diaryDir, { recursive: true });
        console.log('📁 已创建 diary 目录');
    }
    
    // 生成日记内容
    const content = generateDiaryContent(dayNum, customContent);
    const diaryPath = path.join(CONFIG.diaryDir, `${today}.md`);
    
    // 写入 Markdown 文件
    fs.writeFileSync(diaryPath, content, 'utf-8');
    console.log(`✅ 已生成日记: ${diaryPath}`);
    
    // 更新索引
    const data = updateDataIndex(today, `# ${title}`, content);
    
    // 输出 JSON 结果供其他脚本使用
    const result = {
        success: true,
        date: today,
        dayNumber: dayNum,
        file: diaryPath,
        dataFile: CONFIG.dataFile,
        totalDiaries: data.diaries.length
    };
    
    console.log('');
    console.log('---');
    console.log(JSON.stringify(result, null, 2));
    
    return result;
}

// 执行
try {
    main();
} catch (error) {
    console.error('❌ 错误:', error.message);
    process.exit(1);
}